File: backend-kotlin/src/main/kotlin/pl/zycienakodach/eventmodeling/githubclone/slices/read/pullrequestcomments/infrastructure/InMemoryPullRequestCommentsRepository.kt
package pl.zycienakodach.eventmodeling.githubclone.slices.read.pullrequestcomments.infrastructure

import pl.zycienakodach.eventmodeling.githubclone.slices.read.pullrequestcomments.application.PullRequestComment
import pl.zycienakodach.eventmodeling.githubclone.slices.read.pullrequestcomments.application.PullRequestCommentsRepository
import java.util.concurrent.ConcurrentHashMap

class InMemoryPullRequestCommentsRepository : PullRequestCommentsRepository {

    private val entries = ConcurrentHashMap<String, PullRequestComment>()

    override fun findAllBy(pullRequestId: String): List<PullRequestComment> {
        return entries.values.filter { it.pullRequestId == pullRequestId }
    }

    override fun save(comment: PullRequestComment) {
        entries[comment.commentId] = comment
    }
}

File: backend-dotnet/EventModelingGitHubCloneDotNet/EventModelingGitHubCloneDotNet/Slices/Read/PullRequestComments/Infrastructure/InMemoryPullRequestCommentsRepository.cs
using System.Collections.Concurrent;
using System.Collections.Immutable;
using System.Linq;
using System.Threading.Tasks;
using EventModelingGitHubCloneDotNet.Slices.Read.PullRequestComments.Core.Application;

namespace EventModelingGitHubCloneDotNet.Slices.Read.PullRequestComments.Infrastructure
{
    class InMemoryPullRequestCommentsRepository : IPullRequestCommentsRepository
    {
        private readonly ConcurrentDictionary<string, PullRequestComment> _entities =
            new ConcurrentDictionary<string, PullRequestComment>();

        public Task Save(PullRequestComment comment)
        {
            _entities[comment.CommentId] = comment;
            return Task.CompletedTask;
        }

        public Task<int> CountBy(string pullRequestId)
        {
            return Task.FromResult(_entities.Count);
        }

        public Task<ImmutableList<PullRequestComment>> FindAllBy(string pullRequestId)
        {
            var result = _entities.Values.Where(comment => comment.PullRequestId.Equals(pullRequestId)).ToImmutableList();
            return Task.FromResult(result);
        }
    }
}